plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/install/CommandExecutor.java

```diff
@@ -0,0 +1,101 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * 
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.aps.system.services.digitalexchange.install;
+
+import org.entando.entando.aps.system.services.digitalexchange.install.model.Command;
+import java.io.IOException;
+import java.io.InputStream;
+import javax.xml.bind.JAXB;
+import org.entando.entando.aps.system.services.digitalexchange.install.model.Commands;
+import org.entando.entando.web.common.model.RestResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.BeanInitializationException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.expression.BeanFactoryResolver;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ExpressionParser;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+/**
+ * Executes operations defined in the component.xml using SpEL. The purpose is
+ * to call an arbitrary API controller (even one that is not a dependency of the
+ * DE plugin itself, like CMS controllers). It is also able to interpret some
+ * default commands using the #call custom function. This simplify the
+ * definition of the most common operations inside the component.xml. See the
+ * file commands.xml.
+ */
+@Component
+public class CommandExecutor implements ApplicationContextAware {
+
+    private static final Logger logger = LoggerFactory.getLogger(CommandExecutor.class);
+
+    private final ExpressionParser parser = new SpelExpressionParser();
+    private final StandardEvaluationContext context = new StandardEvaluationContext();
+
+    private final Commands commands;
+
+    public CommandExecutor() {
+
+        try {
+            context.registerFunction("bindingResult", CommandUtils.class.getDeclaredMethod("getBindingResult"));
+            context.registerFunction("call", Command.class.getDeclaredMethod("newInstance", String.class, String[].class));
+        } catch (NoSuchMethodException ex) {
+            throw new BeanInitializationException("Unable to register SpEL functions", ex);
+        }
+
+        try (InputStream in = getClass().getClassLoader().getResourceAsStream("component/plugins/digitalExchange/commands.xml")) {
+            commands = JAXB.unmarshal(in, Commands.class);
```

Comment:
In this line the input stream (in) can be null. Maybe treat it or log something

Reply:
Done